diff --git a/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/__init__.py b/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/__init__.py
index ece0ab7312..6f78baf1b6 100644
--- a/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/__init__.py
+++ b/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/__init__.py
@@ -220,10 +220,16 @@ def get_contents_dir(node):
 def get_contents_file(node):
     if not node.rexists():
         return b''
+
+    cached = getattr(node.attributes, "contents", None)
+    if cached is not None:
+        return cached
+
     fname = node.rfile().get_abspath()
     try:
         with open(fname, "rb") as fp:
             contents = fp.read()
+            setattr(node.attributes, "contents", contents)
     except EnvironmentError as e:
         if not e.filename:
             e.filename = fname
