diff --git a/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/FS.py b/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/FS.py
index 352aeaebcc..e02a1c4e38 100644
--- a/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/FS.py
+++ b/src/third_party/scons-3.0.4/scons-local-3.0.4/SCons/Node/FS.py
@@ -2643,6 +2643,16 @@ class File(Base):
         Base.__init__(self, name, directory, fs)
         self._morph()
 
+        try:
+            ssign = self.dir.sconsign()
+            if ssign is None:
+                raise KeyError
+            stored_info = ssign.get_entry(self.name)
+            self.ninfo = stored_info.ninfo
+            self.binfo = stored_info.binfo
+        except (KeyError, EnvironmentError):
+            self.ninfo = self.new_ninfo()
+
     def Entry(self, name):
         """Create an entry node named 'name' relative to
         the directory of this file."""
